SRVKE-695: Updated event source descriptions (#34583)
openshift/openshift-docs: serverless/event_sources/knative-event-sources.adoc

include::modules/serverless-document-attributes.adoc[]
[id="knative-event-sources"]
= Understanding event sources
include::modules/common-attributes.adoc[]
:context: knative-event-sources

toc::[]

> A Knative _event source_ can be any Kubernetes object that generates or imports CloudEvents, and relays those events to another endpoint, known as a xref:../../serverless/knative_eventing/serverless-event-sinks.adoc#serverless-event-sinks[_sink_]. Sourcing events is critical to developing a distributed system that reacts to events.

Reviewer: would it be worth perhaps rephrasing as:
`A Knative _event source_ can be any Kubernentes object that imports data and translates and/or generates CloudEvents...` ?  My only fear with the current wording is that it might imply the existing data source (say, the apiserver itself) actually generates the CloudEvent, vs the source that does it?

Author: @matzew wrote this bit so let's ask him 🙂

Reviewer: Ah ok, so we're grabbing this text from elsewhere? fine by me then

> A Knative _event source_ can be any Kubernetes object that generates or imports CloudEvents, and relays those events to another endpoint, known as a xref:../../serverless/knative_eventing/serverless-event-sinks.adoc#serverless-event-sinks[_sink_]. Sourcing events is critical to developing a distributed system that reacts to events.

Reviewer 2: Could CloudEvents become something like "`CloudEvent` objects" or something like that? IIRC, our in-house guidelines discourage using terms like that as nouns.

Author: There were a few other places this was used in the docs, so I've updated to either "cloud events" or `CloudEvent` object as appropriate. FWIW I really don't like this guideline because it makes our docs a lot clunkier to read but I guess while we're part of OCP docs we're tied to it.

> A Knative _event source_ can be any Kubernetes object that generates or imports CloudEvents, and relays those events to another endpoint, known as a xref:../../serverless/knative_eventing/serverless-event-sinks.adoc#serverless-event-sinks[_sink_]. Sourcing events is critical to developing a distributed system that reacts to events.

Reviewer 2: ```suggestion
A Knative _event source_ can be any Kubernetes object that generates or imports CloudEvents. Event sources relay those events to an endpoint, which is known as a xref:../../serverless/knative_eventing/serverless-event-sinks.adoc#serverless-event-sinks[_sink_]. Sourcing events is critical to developing a distributed system that reacts to events.
```
Just a thought about breaking this up.

Author: I kind of like this the way it is if that's OK, I think it's a better flow than breaking it up.

> You can create and manage Knative event sources using the *Developer* perspective in the {product-title} web console, the `kn` CLI, or by applying YAML files.

Reviewer 2: ```suggestion
You can create and manage Knative event sources by using the *Developer* perspective in the {product-title} web console, the `kn` CLI, or by applying YAML files.
```
